### PR TITLE
scripts: let the user know if submitting with sbatch failed

### PR DESCRIPTION
**Pull request by albestro**
Currently the script submitting benchmark jobs is not reporting anything from the sbatch submission, resulting in the user now knowing if the job has been queued or not.

With this PR, in case of error the output is reported to the user("soft failure", just error reported without making the launching script to terminate).

```
(dla-future)[todi][ialberto@todi-ln001 dla-future.master]$ python3 scripts/gen-dlaf-strong-panel-workers.py
Submitting : sbatch --chdir=/capstor/scratch/cscs/ialberto/20241204-dlaf-strong-panel-workers-local-40/d/0.25 /capstor/scratch/cscs/ialberto/20241204-dlaf-strong-panel-workers-local-40/d/0.25/job_dlaf.sh using shell /usr/local/bin/bash
Submitting : sbatch --chdir=/capstor/scratch/cscs/ialberto/20241204-dlaf-strong-panel-workers-local-50/d/0.25 /capstor/scratch/cscs/ialberto/20241204-dlaf-strong-panel-workers-local-50/d/0.25/job_dlaf.sh using shell /usr/local/bin/bash
Job submission failed with return code 1
sbatch: error: QOSMaxSubmitJobPerUserLimit
sbatch: error: Batch job submission failed: Job violates accounting/QOS policy (job submit limit, user's size and/or time limits)

```

Any suggestions/preference is welcome. Some possible discussion points:
- should we report the error code at all?
- add a success message when everything works

### CODE REVIEW

**Comment by albestro** [comment]
After a chat with @rasolca about this, I slightly changed the behaviour.

@rasolca recalled that there are two use-cases:
- submitting via sbatch
- launching via bash script (e.g. local runs without slurm queue)

This PR is trying to address job submission to the queue, which does not happen in the latter case.

- Using it with slurm will end up printing "submitting" and either nothing more or the full `sbatch` output in case of error.
- Using it in local (i.e. without queuing) it will print out the full output, both stdout and stderr.

In case of error, either from the sbatch queueing command or the bash script (e.g. just last command failing?), the return code is checked and a failure message is printed.

Waiting for your comments for last refinements.

File: scripts/miniapps.py [hunk 1]
```diff
@@ -157,6 +157,13 @@ def submitJobs(self, run_dir, debug=False):
             stdout=PIPE,
             stderr=PIPE,
         )
+
+        raw_stdout, raw_stderr = process.communicate()
+        print(raw_stdout, end="")
+        print(raw_stderr, end="")
+        if process.returncode != 0:
+            print(f"Job submission FAILED.")
```
Comment (msimberg) [inline scripts/miniapps.py:165]: In the case that one is submitting locally without slurm, will this mean that if there's an actual miniapp failure, the script will stop submitting jobs? That might not be the desired behaviour, but then I never use that so I'm not sure what the desired behaviour is.

@biddisco's input would be very useful here.

**Review by msimberg — COMMENTED** [review]
Thanks @albestro for taking this on. I've also suffered in the past from errors not being reported.

File: scripts/miniapps.py (hunk 1 above)
Comment (albestro) [reply scripts/miniapps.py:165]: No, actually we just report the output/error but we don't stop in any case the execution.

I never used it locally, so I might be missing something. In the local case, the return code depends on the last command executed by the bash script, and just if the last one fails the "FAILED" message is printed, but no "FAILED" printed if any of the "middle" runs fails.

Comment (msimberg) [reply scripts/miniapps.py:165]: Ah, good point, I misread the code... In that case, probably no concern here from my side.

**Review by msimberg — APPROVED** [review]

**Review by RMeli — APPROVED** [review]

**Pull request merged**